src/backend/curses/pan.rs

```diff
@@ -0,0 +1,218 @@
+        f();
+        self.window.attroff(style);
+        // pancurses::attron(current_style);
+    }
+
+    fn with_effect<F: FnOnce()>(&self, effect: Effect, f: F) {
+        let style = match effect {
+            // A_REVERSE, taken from ncurses
+            Effect::Reverse => 1 << (10 + 8u32),
+            Effect::Simple => pancurses::A_NORMAL,
+        };
+        self.window.attron(style);
+        f();
+        self.window.attroff(style);
+    }
+
+    fn clear(&self) {
+        self.window.clear();
+    }
+
+    fn refresh(&mut self) {
+        self.window.refresh();
+    }
+
+    fn print_at(&self, (x, y): (usize, usize), text: &str) {
+        self.window.mvaddstr(y as i32, x as i32, text);
+    }
+
+    fn poll_event(&self) -> Event {
+        // TODO: there seems to not be any indication
+        // of Ctrl/Alt/Shift in these :v
+        if let Some(ev) = self.window.getch() {
+            match ev {
+                pancurses::Input::Character(c) => Event::Char(c),
+                pancurses::Input::Unknown(i) => Event::Unknown(i),
+                // TODO: I honestly have no fucking idea what KeyCodeYes is
+                pancurses::Input::KeyCodeYes => Event::Refresh,
+                pancurses::Input::KeyBreak => Event::Key(Key::PauseBreak),
+                pancurses::Input::KeyDown => Event::Key(Key::Down),
+                pancurses::Input::KeyUp => Event::Key(Key::Up),
+                pancurses::Input::KeyLeft => Event::Key(Key::Left),
+                pancurses::Input::KeyRight => Event::Key(Key::Right),
+                pancurses::Input::KeyHome => Event::Key(Key::Home),
+                pancurses::Input::KeyBackspace => Event::Key(Key::Backspace),
+                pancurses::Input::KeyF0 => Event::Key(Key::F0),
+                pancurses::Input::KeyF1 => Event::Key(Key::F1),
+                pancurses::Input::KeyF2 => Event::Key(Key::F2),
+                pancurses::Input::KeyF3 => Event::Key(Key::F3),
+                pancurses::Input::KeyF4 => Event::Key(Key::F4),
+                pancurses::Input::KeyF5 => Event::Key(Key::F5),
+                pancurses::Input::KeyF6 => Event::Key(Key::F6),
+                pancurses::Input::KeyF7 => Event::Key(Key::F7),
+                pancurses::Input::KeyF8 => Event::Key(Key::F8),
+                pancurses::Input::KeyF9 => Event::Key(Key::F9),
+                pancurses::Input::KeyF10 => Event::Key(Key::F10),
+                pancurses::Input::KeyF11 => Event::Key(Key::F11),
+                pancurses::Input::KeyF12 => Event::Key(Key::F12),
+                pancurses::Input::KeyF13 => Event::Shift(Key::F1),
+                pancurses::Input::KeyF14 => Event::Shift(Key::F2),
+                pancurses::Input::KeyF15 => Event::Shift(Key::F3),
+                // TODO: not sure what those are
+                pancurses::Input::KeyDL => Event::Refresh,
+                pancurses::Input::KeyIL => Event::Refresh,
+                pancurses::Input::KeyDC => Event::Refresh,
```

Comment:
`KeyDC` seems to correspond to the `Del` key.
`KeyIC` would be the `Insert` key.